src/components/DailyRoutinePage/DailyRoutine.js

```diff
@@ -1,35 +1,84 @@
 import "./DailyRoutine.css";
 import AddedProductCard from "./AddedProductCard.js";
 import { FaArrowLeft } from "react-icons/fa";
+import { useParams, useHistory } from "react-router-dom";
+import { getDataFromLocalStorage } from "../../utility/localStorage.js";
+import { useEffect, useState } from "react";
 
 export default function DailyRoutine() {
+  const { weekday } = useParams();
+  const history = useHistory();
+  const [allItems, setAllItems] = useState([]);
+  const [products, setProducts] = useState([]);
+
+  useEffect(() => {
+    const fetchProducts = () => {
+      return fetch("/products.json")
+        .then((response) => response.json())
+        .then((productData) => {
+          setProducts(productData);
+        });
```

Comment:
You should add a `.catch` callback, in case something goes wrong. This could set an error state to true, and then you could render an error page or redirect the user to another page

Reply:
adding this to my backlog 👌